ranger/default.go

```diff
@@ -0,0 +1,390 @@
+		return func() error {
+			switch rng.Env {
+			case Testing: // NOTE(dlk): this is an unexpected case since go test does not reach this
+				cfg = &postgres.CxnConfig{
+					Host:     envVarOrString(dbTestHostEnvVar, defaultDBTestHost),
+					IsTestDB: true,
+					Name:     os.Getenv(dbTestNameEnvVar),
+					Password: os.Getenv(dbTestPassEnvVar),
+					Port:     envVarOrString(dbTestPortEnvVar, defaultDBTestPort),
+					User:     os.Getenv(dbTestUserEnvVar),
+				}
+
+			default:
+				if url := os.Getenv(dbURLEnvVar); url != "" {
+					cfg = &postgres.CxnConfig{IsTestDB: false, URL: url}
+				} else {
+					cfg = &postgres.CxnConfig{
+						Host:     envVarOrString(dbHostEnvVar, defaultDBHost),
+						IsTestDB: false,
+						Name:     os.Getenv(dbNameEnvVar),
+						Password: os.Getenv(dbPassEnvVar),
+						Port:     envVarOrString(dbPortEnvVar, defaultDBPort),
+						User:     os.Getenv(dbUserEnvVar),
+					}
+				}
+			}
+
+			db, err := postgres.Connect(cfg, list)
+			if err != nil {
+				return err
+			}
+
+			fn, err := WithDB(postgres.NewService(db))(rng)
+			if err != nil {
+				return err
+			}
+
+			return fn()
+		}, nil
+	}
+}
+
+// DefaultKeyring constructs a RangerOption that applies the default context keys
+// and those keys passed in to the Ranger.
+func DefaultKeyring(keys ...keyring.Keyable) RangerOption {
+	return func(rng *Ranger) (optFollowup, error) {
+		kr := keyring.NewKeyring(
+			defaultSessionCtxKey,
+			defaultCurrentUserCtxKey,
+			append(keys, defaultRequestIDCtxKey)...,
+		)
+
+		return WithKeyring(kr)(rng)
+	}
+}
+
+// DefaultLogger constructs a RangerOption that applies the default logger (logger.DefaultLogger)
+// to the Ranger.
+func DefaultLogger(opts ...logger.LoggerOptFn) RangerOption {
+	logLvl := envVarOrLogLevel(logLevelEnvVar, logger.LogLevelInfo)
+	args := []logger.LoggerOptFn{
+		logger.WithLevel(logLvl),
+	}
+	for _, opt := range opts {
+		args = append(args, opt)
+	}
+
+	return func(rng *Ranger) (optFollowup, error) {
+		return WithLogger(logger.NewLogger(args...))(rng)
+	}
+}
+
+// DefaultParser constructs a RangerOption that configures a default HTML template parser to be used
+// when responding to HTTP requests with *resp.Responder.Html.
+//
+// BASE_URL ought to be set when the default http://localhost:3000 is not wanted.
+//
+// DefaultParser makes available these functions in an HTML template:
+//
+// - "env"
+// - "nonce"
+// - "rootUrl"
+// - "packTag"
+// - "isDevelopment"
+// - "isStaging"
+// - "isProduction"
+func DefaultParser(files fs.FS, opts ...template.ParserOptFn) RangerOption {
+	u, err := url.ParseRequestURI(envVarOrString(baseURLEnvVar, defaultBaseURL))
+	if err != nil {
+		return func(_ *Ranger) (optFollowup, error) {
+			return nil, fmt.Errorf("%w: failed parsing %s: %s", ErrBadConfig, baseURLEnvVar, err)
+		}
+	}
+
+	return func(rng *Ranger) (optFollowup, error) {
+		args := []template.ParserOptFn{
+			template.WithFn(template.Env(rng.Env.String())),
```

Comment:
If we end up supporting server rendered components, might be nice to include Masterminds/sprig here or something similar.